app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachment/AttachmentOptions.kt

```diff
@@ -9,27 +9,50 @@ import androidx.compose.foundation.layout.PaddingValues
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.SnackbarHostState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.wire.android.R
 import com.wire.android.appLogger
 import com.wire.android.ui.common.AttachmentButton
 import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.home.conversations.AttachmentPart
 import com.wire.android.util.permission.UseCameraRequestFlow
+import com.wire.android.util.permission.UseStorageRequestFlow
 import com.wire.android.util.permission.rememberCaptureVideoFlow
 import com.wire.android.util.permission.rememberCurrentLocationFlow
 import com.wire.android.util.permission.rememberOpenFileBrowserFlow
 import com.wire.android.util.permission.rememberOpenGalleryFlow
 import com.wire.android.util.permission.rememberRecordAudioRequestFlow
 import com.wire.android.util.permission.rememberTakePictureFlow
 
-@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class)
+@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class, ExperimentalMaterial3Api::class)
 @Composable
-fun AttachmentOptionsComponent() {
-    val attachmentOptions = buildAttachmentOptionItems()
+fun AttachmentOptionsComponent(onSendAttachment: (AttachmentPart?) -> Unit, snackbarHostState: SnackbarHostState) {
+    val viewModel: AttachmentOptionsViewModel = hiltViewModel()
+    val attachmentOptions = buildAttachmentOptionItems(viewModel)
+
+    // handle view states
+    when (val state = viewModel.attachmentState) {
+        is AttachmentState.Initial -> appLogger.d("Not picked yet")
+        is AttachmentState.Picked -> onSendAttachment(state.attachmentPart)
+        is AttachmentState.Error -> {
+            // FIXME. later on expand to other possible errors
+            val errorMessage = stringResource(R.string.error_unknown_message)
+            LaunchedEffect(state) {
+                snackbarHostState.showSnackbar(errorMessage)
+                viewModel.resetViewState()
+            }
+        }
+    }
```

Comment:
Can we extract this to a separated function and call it inside the `init { }` block?

Reply:
I'm refactoring this states and moving away from the viewmodel approach, so I think this will be outdated code and it'll change a lot

Reply:
This survived the refactor, but I only extracted the logic to a separate function, because t it's not possible to move it to an `init block` since this is not a class, just a file with composables